site/sfguides/src/gain_insights_from_unstructured_data/gain_insights_from_unstructured_data.md

```diff
@@ -0,0 +1,466 @@
+                                                              F.lit('Unsure')))
+
+  reviews_df.select(["REVIEW","CLEAN_RECOMMEND"]).show(3)
+  ```
+<!-- ------------------------ -->
+
+## Extract Answers from what your customers are saying
+Duration: 5
+
+### Overview
+
+In this section, you will leverage **Snowflake Cortex - Extract Answer** to get answers to your questrions:
+
+### Answer specifc quesions you have 
+
+* Using **Snowflake Cortex - Extract Answer** to dive into quesitons you have
+
+  ```python
+  # Step 1: Add a row number for each review within each TRUCK_BRAND_NAME
+  window_spec = Window.partition_by("TRUCK_BRAND_NAME").order_by("REVIEW")
+  ranked_reviews_df = reviews_df.with_column(
+      "ROW_NUM", F.row_number().over(window_spec)
+  )
+
+  # Step 2: Filter to include only the first 20 rows per TRUCK_BRAND_NAME to get a general idea
+  filtered_reviews_df = ranked_reviews_df.filter(F.col("ROW_NUM") <= 20)
+
+  # Step 3: Aggregate reviews by TRUCK_BRAND_NAME
+  aggregated_reviews_df = filtered_reviews_df.group_by("TRUCK_BRAND_NAME").agg(
+      F.array_agg(F.col("REVIEW")).alias("ALL_REVIEWS")
+  )
+
+  # Step 4: Convert the array of reviews to a single string
+  concatenated_reviews_df = aggregated_reviews_df.with_column(
+      "ALL_REVIEWS_TEXT", F.call_function("array_to_string", F.col("ALL_REVIEWS"), F.lit(' '))
+  )
+
+  # Step 5: Generate summaries for each truck brand
+  summarized_reviews_df = concatenated_reviews_df.with_column(
+      "NUMBER_ONE_DISH", cortex.ExtractAnswer(F.col("ALL_REVIEWS_TEXT"), "What is the number one dish positivly mentioned in the feedback?")
+  )
+
+  # Step 6: Extract the first element of the array
+  first_element_df = summarized_reviews_df.with_column(
+      "FIRST_ELEMENT", F.expr("NUMBER_ONE_DISH[0]")
+  )
+
+  # Step 7: Parse the first element as JSON and extract the "answer" field
+  readable_df = first_element_df.with_column(
+      "NUMBER_ONE_DISH", F.get(F.parse_json(F.col("FIRST_ELEMENT")), F.lit("answer"))
+  )
+
+  # Display the simplified results
+  readable_df.select(["TRUCK_BRAND_NAME", "NUMBER_ONE_DISH"]).show()
+  ```
+<!-- ------------------------ -->
+
+## Understand customer sentiment 
+Duration: 2
+
+### Enabled by Cortex Sentiment
+
+So far you saw Snowflake Cortex - Translate & Complete. Next, you will look at another **task specific LLM function in Cortex - Sentiment**. This sentiment function is used to understand the customer's tone  based on the review they provided. Sentiment return value between -1 and 1 such that -1 is the most negative while 1 is the most positive. This is done within the notebook using the following code snippet in cell `CORTEX_SENTIMENT`.
+
+```python
+# Understand the sentiment of customer review using Cortex Sentiment
+reviews_df = reviews_df.withColumn('SENTIMENT', cortex.Sentiment(F.col('REVIEW')))
+
+reviews_df.select(["REVIEW","SENTIMENT"]).show(3)
+```
+
+<!-- ------------------------ -->
+## Conclusion
```

Comment:
@jamescha-earley Can you rename this to "Conclusion And Resources"?

Reply:
Done